src/action/timeline.rs

```diff
@@ -103,19 +107,33 @@ impl RecurseState {
     where
         S: Session,
     {
-        let metadata = match symlink_metadata(&root) {
-            Ok(metadata) => metadata,
-            Err(_) => return Ok(()),
-        };
-        let entry = entry_from_metadata(&metadata, root);
-        self.process_entry(entry, session)?;
-        if metadata.is_dir() && metadata.dev() == self.device {
-            let entry_iter = match read_dir(root) {
-                Ok(iter) => iter,
-                Err(_) => return Ok(()),
+        let mut path = root.clone();
+        let mut dir_iter_stack = Vec::new();
+        loop {
```

Comment:
Is there a reason why you decided to switch from normal recursion to implementing recursion stack yourself? I think the previous code read better and felt more natural.

Reply:
the previous code stackoverflow'd in case of deep directory trees and i had no idea how to fix that without getting rid of recursion